components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -567,24 +567,28 @@ func runIDEReadinessProbe(cfg *Config) {
 	case ReadinessHTTPProbe:
 		var (
 			url    = fmt.Sprintf("http://localhost:%d/%s", cfg.IDEPort, strings.TrimPrefix(cfg.ReadinessProbe.HTTPProbe.Path, "/"))
-			client = http.Client{Timeout: 5 * time.Second}
-			tick   = time.NewTicker(1 * time.Second)
+			client = http.Client{Timeout: 1 * time.Second}
+			tick   = time.NewTicker(500 * time.Millisecond)
 		)
 		defer tick.Stop()
+
+		t0 := time.Now()
+
 		for {
+			<-tick.C
```

Comment:
placing the tick here means we might be 500 milliseconds slower than we need to be if the IDE is already available by the time we start probing.

Reply:
yes, I am aware :)